hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/services/RpAuthnResponseGeneratorService.java

```diff
@@ -36,13 +39,16 @@ public AuthnResponseFromHubContainerDto generate(ResponseFromHubDto responseFrom
     private AuthnResponseFromHubContainerDto createSuccessResponse(final ResponseFromHubDto responseFromHub) {
         String authnRequestIssuerEntityId = responseFromHub.getAuthnRequestIssuerEntityId();
 
+        List<String> encryptedAssertions = new ArrayList<>();
+        responseFromHub.getEncryptedMatchingServiceAssertion().ifPresent(encryptedAssertions::add);
```

Comment:
You could also do `List<String> encryptedAssertions = responseFromHub.getEncrypterMatchingDatasetAssertion().map(Collections::singletonList).orElse(Collections.emptyList())` but given this code is only going to exist for 2 releases there is no point nitpicking

Reply:
That's nicer.  I'm always happy to get rid of imperative code.